Cpp_hands_on_programming/lib/abstract/Ipc_method.cc

```diff
@@ -0,0 +1,60 @@
+#include "Ipc_method.h"
+#include <iostream>
+
+int Ipc_method::getFileSize(std::string filename) {
+    int size;
+    std::ifstream in(filename, std::ios::binary);
+	if (!in) {
+		perror("getFileSize");
+		exit(EXIT_FAILURE);
+	}
+	in.seekg(0, std::ios::end);
```

Comment:
Tbh I am not 100% convinced by the document, I have tested the previous C and this C++ program with both text and binary files, it works fine for me. But I could have missed some cases that would cause some problem? Have you tested the cases which error occur using seek and tell?

Reply:
No I don't.

I have found this : https://stackoverflow.com/a/22986486

And some elements: ```he standard doesn't require tellg to return the byte offset from the beginning of the file. In general, this may not be a reliable way to get the size of the file, though it probably does what you expect on Linux and Windows.```

Reply:
Ok it is safe to assume it could behave differently in some unknown cases. I will probably use some other functions to get the file size, no problem